### PR TITLE
Add Ruby RBS Type Signatures 

### PR DESCRIPTION
**Pull request by mattt**
This PR explores the use of [RBS](https://github.com/ruby/rbs) and [Steep](https://github.com/soutaro/steep) to improve our Ruby source code.

Run the following commands in this branch to try it out:

```terminal
$ cd common
$ bundle install
$ bundle exec steep check
```

### CODE REVIEW

**Comment by mattt** [comment]
After doing the work to implement type signatures manually for a small ecosystem, I've concluded that it's too much effort. If we could get RBS / Steep to infer types automatically (perhaps by co-mingling code that calls ecosystem plugins directly, like in [dependabot-script](https://github.com/dependabot/dependabot-script)), then it might be worth taking another look.

File: common/lib/dependabot/file_fetchers/base.rb [hunk 1]
```diff
@@ -61,10 +66,17 @@ def target_branch
...
+      # the repo this instance was created with.
+      # Returns an array of `Dependabot::DependencyFile` instances.
       def files
         @files ||= fetch_files
       end
 
+      # Returns the commit SHA-1 hash at the time
+      # the dependency files were fetched.
+      # If called before files, the returned value
+      # will be used in subsequent calls to files.
```
Comment (jeffwidman) [inline common/lib/dependabot/file_fetchers/base.rb:79]: @mattt Do you want to pull these docstrings out into a separate PR and add them? They look useful...
